harvest/storage/database_storage.py

```diff
@@ -68,21 +56,81 @@ def store(self, symbol: str, interval: str, data: pd.DataFrame, remove_duplicate
         :data: a pandas dataframe that has stock data and has a datetime 
             index
         """
-        super().store(symbol, interval, data, remove_duplicate)
 
         if not data.empty:
-            self.storage_lock.acquire()
-
-            df = self.storage[symbol][interval]
-            df.columns = [column[1] for column in df.columns]
-            df.rename(columns={'open': 'open_'}, inplace=True)
-            df['timestamp'] = df.index 
-            df['symbol'] = symbol
-            df['interval'] = interval
-            data = df.to_dict(records)
-
-            with Session.begin() as session:
-                session.add_all([Asset(**d) for d in data])
+
+            data.index = normalize_pandas_dt_index(data)
+            data.columns = [column[1] for column in data.columns]
+            data.rename(columns={'open': 'open_'}, inplace=True)
+            data['timestamp'] = data.index 
+            data['symbol'] = symbol
+            data['interval'] = interval
+            data = data.to_dict('records')
+
+            with self.Session.begin() as session:
+                [session.merge(Asset(**d)) for d in data]
                 session.commit()
 
-            self.storage_lock.release()
\ No newline at end of file
+    def aggregate(self, symbol: str, base: str, target: str, remove_duplicate: bool=True):
+        """
+        Aggregates the stock data from the interval specified in 'from' to 'to'.
+        """
+
+        data = self.load(symbol, base)
+        agg_data = self._append(self.load(symbol, target), aggregate_df(data, target), remove_duplicate)
```

Comment:
There is a bug in this line where if the `target` interval does not exist, this function will fail.